dj_urls_panel/views.py

```
@@ -99,12 +157,255 @@ def url_detail(request, pattern):
+                "headers": response_headers,
+                "body": response_body,
+                "is_json": is_json,
+                "elapsed_ms": int(response.elapsed.total_seconds() * 1000),
+                "url": response.url,  # Final URL after redirects
+            }
+        )
+
+    except http_requests.exceptions.Timeout:
+        return JsonResponse({"error": "Request timed out"}, status=408)
+    except http_requests.exceptions.ConnectionError as e:
+        return JsonResponse({"error": f"Connection error: {str(e)}"}, status=502)
```

Ruff (0.14.14)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)